azplugins/SinusoidalChannelGeometry.h

```diff
@@ -0,0 +1,366 @@
+                    }
+
+            }
+
+
+            // Remaining integration time dt is amount of time spent traveling distance out of bounds.
+            dt = fast::sqrt(((pos.x - x0)*(pos.x - x0) + (pos.y - y0)*(pos.y -y0) + (pos.z - z0)*(pos.z - z0))/(vel.x*vel.x + vel.y*vel.y + vel.z*vel.z));
+
+            //  positions are updated
+            pos.x = x0;
+            pos.y = y0;
+            pos.z = z0;
+
+            /* update velocity according to boundary conditions.
+             *
+             * A upwards normal of the surface is given by (-df/dx,-df/dy,1) with f = (A*cos(x*2*pi*p/L) +/- sign*h), so
+             * normal  = (A*2*pi*p/L*sin(x*2*pi*p/L),0,1)/|length|
+             * The direction of the normal is not important for the reflection.
+             * Calculate components by hand to avoid sqrt in normalization of the normal of the surface.
+             */
+            Scalar3 vel_new;
+            if (m_bc ==  mpcd::detail::boundary::no_slip) // No-slip requires reflection of both tangential and normal components:
+                {
+
+                vel_new.x = -vel.x;
+                vel_new.y = -vel.y;
+                vel_new.z = -vel.z;
+
+                }
+            else // Slip conditions require only tangential components to be reflected:
+                {
+                Scalar B = m_Amplitude*m_pi_period_div_L*fast::sin(x0*m_pi_period_div_L);
+
+                vel_new.x = vel.x - 2*B*(B*vel.x + vel.z)/(B*B+1);
+                vel_new.y = vel.y;
+                vel_new.z = vel.z - 2*(B*vel.x + vel.z)/(B*B+1);
+                }
+
+            vel = vel_new;
+
+            return true;
+            }
+
+        //! Check if a particle is out of bounds
+        /*!
+         * \param pos Current particle position
+         * \returns True if particle is out of bounds, and false otherwise
+         */
+        HOSTDEVICE bool isOutside(const Scalar3& pos) const
+            {
+            Scalar a = pos.z - m_Amplitude*fast::cos(pos.x*m_pi_period_div_L);
+            return (a > m_h || a < -m_h);
+            }
+
+        //! Validate that the simulation box is large enough for the geometry
+        /*!
+         * \param box Global simulation box
+         * \param cell_size Size of MPCD cell
+         *
+         * The box is large enough for the cosine if it is padded along the z direction so that
+         * the cells just outside the highest point of the cosine + the filler thinckness
+         * would not interact with each other through the boundary.
+         *
+         */
+        HOSTDEVICE bool validateBox(const BoxDim& box, Scalar cell_size) const
+            {
+            const Scalar hi = box.getHi().z;
+            const Scalar lo = box.getLo().z;
+            // TO DO: get max_shift from  mpcd , can't use function argument because hoomd/mpcd/ConfinedStreamingMethod.h complains
+            // if arguments of validateBox() change
```

Comment:
How did I deal with this issue in the slit filler? (or didn't I?)

Reply:
You did not have to deal with this, you  simply said the box needs to be padded by 1 cell size, see here: https://github.com/glotzerlab/hoomd-blue/blob/ed60448e7720fe6bade26ca53876259d93eec9ea/hoomd/mpcd/SlitGeometry.h#L148